### ISSUE TITLE
Cannot delete spaces. "500 Internal Server Error" from the API

### ISSUE THREAD

**Issue by rgarg1** [openshiftio/openshift.io#3503]
1) Log into OSIO
2) Browse to the page to see all your spaces
3) Try deleting a space. Observe that space is not deleted and you see a 500 error response from the backend API

![cannot delete space](https://user-images.githubusercontent.com/28748448/39742774-633ce658-52bc-11e8-84e7-312ffa357516.png)

**Comment by rgarg1** [comment]
Priority @qodfathr ?

**Comment by kwk** [comment]
I can confirm the issue but interestingly the space disappears from the list of my spaces anyways.

**Comment by joshuawilson** [comment]
Is this a refresh issue? or does not delete it all?

**Comment by alexeykazakov** [comment]
Maybe related to https://github.com/openshiftio/openshift.io/issues/3580

**Comment by ldimaggi** [comment]
See: #3580 

The error is due to the spaces including characters that the UI prevents the users from including in space names. 

@rgarg1  To confirm - try this with a space (name) that is created in the UI.

**Comment by aptmac** [comment]
When deleting a space, is the expectation that all of it's data should disappear? I'm not quite sure the act of deleting a space is actually deleting it. When creating a new space the console prints out the updated spaces object, and mine contains a bunch of old "deleted" spaces that I had made for testing a while ago.

Example:

I currently only have 4 spaces as seen on the 'My Spaces' page:
![myspaces](https://user-images.githubusercontent.com/10425301/40368456-d7f657c0-5da9-11e8-997c-3caf855566f6.png)

When I create a new space, the console logs the updated spaces object. It still has record of all the work spaces that I created (and deleted).
![console1](https://user-images.githubusercontent.com/10425301/40368935-d7668112-5daa-11e8-8fb4-323fc5b76f4d.png)

I actually had so much text in that object that the console asked me if I wanted to view the next 25,000 characters.
![console2](https://user-images.githubusercontent.com/10425301/40368458-d813cc92-5da9-11e8-8b36-a79ebfadb049.png)

**Comment by rgarg1** [comment]
@ldimaggi Yes, I can confirm that spaces created with a **:** in the name have this problem. Though the UI enforces not to have a ':' in the space name, the same is allowed through the API.

**Comment by rgarg1** [comment]
Due to this issue, I'm stuck with piles of test spaces created using API. Reset env is of no help either:
![unable to erase spaces](https://user-images.githubusercontent.com/28748448/40833284-1a04998e-65ab-11e8-93d7-aa42f22870fc.png)

**Comment by joshuawilson** [comment]
@aslakknutsen who can fix this?
